File: study/gyujeong/PGM_42578_LGJ.java
// PGM - 의상
// 해시 및 구현
// https://school.programmers.co.kr/learn/courses/30/lessons/42578


import java.util.*;

public class PGM_42578_LGJ {

    class Solution {
        public int solution(String[][] clothes) {
            int answer = 1;
            HashMap<String, Integer> clothesMap = new HashMap<>();

            for (String[] clothe: clothes) {
                String category = clothe[1];
                if (clothesMap.get(category) == null) clothesMap.put(category, 2);
                else clothesMap.put(category, clothesMap.get(category)+1);
            }
            //System.out.println(clothesMap);

            for (String key: clothesMap.keySet()) {
                answer *= clothesMap.get(key);
            }

            return answer-1;
        }
    }
}

File: assignments/2026/week10/PGM_42578_LGJ.py
# PGM - 42578 의상
# 해쉬
# https://school.programmers.co.kr/learn/courses/30/lessons/42578?language=python3

def solution(clothes):
    answer = 1
    clothes_by_type = {}
    for clothe in clothes:
        clothes_by_type[clothe[1]] = clothes_by_type.get(clothe[1], 1)+1

    for value in clothes_by_type.values():
        answer *= value

    return answer-1